### DISCUSSION TITLE
How to set default login password when deploying automated with kubernetes?

### DISCUSSION THREAD

**Discussion by OmpahDev** [apache/airflow#19687] [Q&A]
I'm deploying an airflow cluster using kubernetes manifests that are run by a CD server. The problem is when I create them, there's no way to login. The documentation says to run a command to create username and password (https://airflow.apache.org/docs/apache-airflow/stable/security/webserver.html), but I can't really do this in my situation, as the actual airflow app is being run on a kubernetes pod that is being automatically spun up. Furthermore, I wouldn't want to have to manually go in and do that anyway, the whole point is to have the setup be all automated.

How can I use kubernetes manifests to pass in a username and password? Something along the lines of adding a parameter in the configmap? I've been combing through documentation but I can't find an answer to these seemingly obvious question. I found information on how to do this when using helm, but I'm not using helm, I'm executing kubernetes manifests directly.

**Comment by OmpahDev** [comment]
I implemented this (successfully) by adding a lifecycle poststart exec on the manifest, but seems a bit janky - there a better way to do this?

**Reply by potiuk** [reply]
I think that's one of the good solutions - if it works for you, then good. 

This is definitely something that you have to figure out how to do in your deployment because every deployment is different. There is no "standard' way  it is left for you to decide how to best deploy it. Some use terraform, some use other ways to get your intial user creation. In many cases airflow is integrated with external authentication which is the most "common" enterprise way of authorisation and authentication.
